mpa/cls/inferrer.py

```diff
@@ -100,7 +100,7 @@ def _infer(self, cfg, dump_features=False, dump_saliency_map=False):
             outputs = data_infos
         else:
             with FeatureVectorHook(model.module.backbone) if dump_features else nullcontext() as fhook:
-                with SaliencyMapHook(model.module.backbone) if dump_saliency_map else nullcontext() as shook:
+                with ActivationMapHook(model.module.backbone) if dump_saliency_map else nullcontext() as shook:
```

Comment:
ASAIK `Inferrer` requires only feature vector for active sampler. `ActivationMapHook` goes to `Explainer`. Based on config, `Explainer` use `ActivationMapHook` or `EigenCamHook` to get appropriate saliency map.

Reply:
Yes, then you mean delete exporting activation map from `Inferrer`?

Reply:
Keep `ActivationMapHook` at the `Inferrer`, temporary. Add `ActivationMapHook` to the `Explainer` as an explain option.

> Based on config, Explainer use ActivationMapHook or EigenCamHook to get appropriate saliency map.